src/trackedit/internal/au3/au3interaction.cpp

```diff
@@ -1141,34 +1202,54 @@ bool Au3Interaction::splitDeleteSelectedOnTracks(const TrackIdList tracksIds, se
     return true;
 }
 
-bool Au3Interaction::trimClipLeft(const ClipKey& clipKey, secs_t deltaSec, bool completed)
+bool Au3Interaction::trimClipsLeft(const ClipKey& clipKey, secs_t deltaSec, secs_t minClipDuration, bool completed)
 {
-    Au3WaveTrack* waveTrack = DomAccessor::findWaveTrack(projectRef(), Au3TrackId(clipKey.trackId));
-    IF_ASSERT_FAILED(waveTrack) {
-        return false;
-    }
+    ClipKeyList clips;
+    if (!muse::contains(selectionController()->selectedClips(), clipKey)) {
+        //! NOTE hover handle single clip trim
+        clips.push_back(clipKey);
+    } else {
+        //! NOTE: handle multi-clip trim
+        //! NOTE: check if trim delta is applicable to every clip
+        std::optional<secs_t> duration = getShortestClipDuration(selectionController()->selectedClips());
 
-    std::shared_ptr<Au3WaveClip> clip = DomAccessor::findWaveClip(waveTrack, clipKey.clipId);
-    IF_ASSERT_FAILED(clip) {
-        return false;
-    }
+        if (!duration.has_value() || !muse::RealIsEqualOrMore(duration.value() - deltaSec, minClipDuration)
+            || !muse::RealIsEqualOrLess(deltaSec, duration.value())
+            || (muse::RealIsEqualOrLess(deltaSec, 0.0) && anyLeftFullyUntrimmed(selectionController()->selectedClips()))) {
+            return false;
+        }
 
-    if (muse::RealIsEqualOrLess(deltaSec, 0.0) && muse::is_equal(clip->GetTrimLeft(), 0.0)) {
-        //! NOTE: clip is fully untrimmed
-        return false;
+        clips = selectionController()->selectedClips();
     }
```

Comment:
I think it's better to split the determination of which clips need to be trimmed and the trim itself into different methods, something like this
```
bool Au3Interaction::trimClipsLeft(const ClipKey& clipKey, secs_t deltaSec, secs_t minClipDuration, bool completed)
{
     ClipKeyList clips = determinateClipsToTrim(...);
     doTrimClipsLeft(clips, ...)  <- private method
}
```